src/test/test_ISComm.cpp

```diff
@@ -1188,3 +1189,147 @@ TEST(ISComm, alternating_isb_nmea_parse_error_check)
     ASSERT_EQ(msgCntNmea, 0);
 }
 #endif
+
+
+#if TEST_TRUNCATED_PACKETS
+TEST(ISComm, IncompletePackets)
+{
+	// Initialize Com Manager
+	init(tcm);
+
+	// Generate and add data to deque
+	generateData(g_testTxDeque);
+
+	int badPktCount = 0, goodPktCount = 0;
+	uint8_t buf[4096] = {0};
+
+	// Use Com Manager to send deque data to Tx port ring buffer
+	for(int i=0; i<g_testTxDeque.size(); i++)
+	{
+		data_holder_t td = g_testTxDeque[i];
+
+		// Send data - writes data to tcm.txBuf
+		int n;
+		switch (td.ptype)
+		{
+		case _PTYPE_INERTIAL_SENSE_DATA:	// IS binary
+		case _PTYPE_INERTIAL_SENSE_CMD:
+			n = is_comm_data_to_buf(buf, sizeof(buf), &g_comm, td.did, td.size, 0, td.data.buf);
+			if (i%4 == 0)
+			{	// Throw away half of packet
+				n = n/2;
+				badPktCount++;
+				DEBUG_PRINTF("[%d] (%d) Bad  %d\n", i, n, badPktCount);
+			}
+			else
+			{
+				goodPktCount++;
+				DEBUG_PRINTF("[%d] (%d) Good %d\n", i, n, goodPktCount);
+			}
+			portWrite(0, buf, n);
+			break;
+
+		case _PTYPE_NMEA:
+		case _PTYPE_UBLOX:
+		case _PTYPE_RTCM3:
+		case _PTYPE_SONY:
+			n = td.size;
+			if (i%4 == 0)
+			{	// Throw away half of packet
+				n = n/2;
+				badPktCount++;
+				DEBUG_PRINTF("[%d] (%d) Bad  %d\n", i, n, badPktCount);
+			}
+			else
+			{
+				goodPktCount++;
+				DEBUG_PRINTF("[%d] (%d) Good %d\n", i, n, goodPktCount);
+			}
+			portWrite(0, td.data.buf, n);
+			break;
+		}
+	}
+
+	// Test that data parsed from Tx port matches deque data
+	is_comm_instance_t &comm = g_comm;
+	protocol_type_t ptype;
+	uDatasets dataWritten;
+	int found=0;
+
+	while (g_testTxDeque.size()>0)
+	{
+		int n = _MIN(ringBufUsed(&tcm.portTxBuf), is_comm_free(&comm));
+		ringBufRead(&tcm.portTxBuf, comm.rxBuf.tail, n);
+
+        // Update comm buffer tail pointer
+        comm.rxBuf.tail += n;
+
+		bool priorBad = false;
+        while ((ptype = is_comm_parse(&comm)) != _PTYPE_NONE) 
+		{
+			if (ptype == _PTYPE_PARSE_ERROR && priorBad)
+			{	// Ignore sequential parse errors
+				continue;
+			}
+
+			data_holder_t td = g_testTxDeque.front();
+			g_testTxDeque.pop_front();
+
+			switch (ptype)
+			{
+			case _PTYPE_INERTIAL_SENSE_DATA:
+				// Found data
+				DEBUG_PRINTF("[%d] (%d) Good %d: ISB %3d\n", found, comm.rxPkt.data.size, g_comm.rxPktCount, comm.rxPkt.hdr.id);
```

Comment:
I don't think I've seen DEBUG_PRINTF before... is this new?  I'm assuming this is only used if its not-embedded?

Reply:
Yes, non-embedded only for testing.  This is a simple way to enable print statements when I need to debug that I don't want to delete.